### PR TITLE
feat: first challenge complete

### PR DESCRIPTION
**Pull request by parra-accedo**
## First Challenge: Core Concepts, Styling, and Testing
- Eliminated unnecessary components on the template and saved some for the future
- Created a Header component with the following: Logo, Searchbar, Theme toggle, Profile thumb (not yet functional)
- Created a Grid view for video cards
- Retrieved data from a mock JSON file and rendered it
- Included Material UI for components (Popover and Switch) and Material Icons

Tests on the following components:
- fns (StringlifyDate): complete functionality checks
- Home page UI: renders properly and doesn't change in Snapshot
- createVideoList function: verified functionality

Code coverage over 90%

NOTE: There is a problem between Jest and the version of React that the template had, so I had to uninstall Jest in order for it to compile. Will make a new project with updated dependencies for next deliverable.

If you want to run the tests, install Jest via:
npm install Jest
And if you want to run the React app, just uninstall it.

Code coverage in Coverage.txt

TODO on next delivery: 
- Make the dropdown popover functional
- Link switch to change themes
- If necessary, create a more precise function to stringlify dates
- Mock test on Toggle component

Captura de Pantalla 2021-03-19 a la(s) 15 05 54

### CODE REVIEW

File: Coverage.txt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+What components are worth to test in your development?
+- Everything, from functions to UI with snapshots
+
+Can you apply TDD once you already created components?
+- Yes, instead of first creating the test and coding, you now see each internal functionality and UI and make tests to see if they are working correctly, even in edge cases
+
+--------------------------------|---------|----------|---------|---------|-------------------
+File                            | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
+--------------------------------|---------|----------|---------|---------|-------------------
+All files                       |   94.03 |    73.91 |   76.47 |   95.45 |
```
Comment (PacoMojica) [inline Coverage.txt:13]: Your code coverage looks good 👍

File: src/components/Header/Header.component.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+`;
+
+const Logo = styled.img`
+  height: 5rem;
+  &:hover {
+    cursor: pointer;
+  }
+`;
+
+const Container = styled.div`
```
Comment (PacoMojica) [inline src/components/Header/Header.component.jsx:32]: This is are well-defined styled components, you can improve this by moving them to their own file, this way when you need to know about the styles of the components you go to that file, if you need to know something about the logic of the component you rea this file and it is not crowded with the styles

File: src/utils/fns.test.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  it('recognizes months correctly', () => {
+    const ONE_MONTH = 1000 * 60 * 60 * 24 * 30; // estimation for a month
+    const today = new Date();
+    const twoMonthsAgo = new Date(today - 2 * ONE_MONTH);
+    const result = fns.stringlifyDate(twoMonthsAgo);
+    expect(result).toBe('Hace 2 meses');
+  });
+
+  it('recognizes hours correctly', () => {
```
Comment (PacoMojica) [inline src/utils/fns.test.js:34]: This are some great tests, you have several tests cases which increases the branch percentage 👌

**Review by PacoMojica — COMMENTED** [review]
Hi Alejandro 👋, your challenges have what was expected for the first deliverable, I like what you have up to this point

Great work!
